sql/binlog.cc

```diff
@@ -534,7 +541,8 @@ class binlog_cache_data {
   int finalize(THD *thd, Log_event *end_event);
   int finalize(THD *thd, Log_event *end_event, XID_STATE *xs);
   int flush(THD *thd, my_off_t *bytes, bool *wrote_xid);
-  int write_event(Log_event *event);
+  int write_event(THD *thd, Log_event *event,
```

Comment:
Even though this comes from the original code, I think there would be fewer changes if we changed this to:

```
int write_event(Log_event *event, THD *thd = nullptr, bool write_meta_data_event = false);
```

since the only reason for passing in the thd is for write_meta_data_event. Either way is fine.